packages/infra/lib/surescripts-stack.ts

```diff
@@ -0,0 +1,527 @@
+      maxMessageCountAlarmThreshold: 15_000,
+      maxReceiveCount: 3,
+      visibilityTimeout: Duration.seconds(sendPatientRequestLambdaTimeout.toSeconds() * 2 + 1),
+      createRetryLambda: false,
+    },
+    eventSource: {
+      batchSize: 1,
+      reportBatchItemFailures: true,
+      maxConcurrency: 4,
+    },
+    waitTime: Duration.seconds(0), // No limit
+  },
+  receiveVerificationResponse: {
+    name: "SurescriptsReceiveVerificationResponse",
+    entry: "surescripts-receive-verification-response",
+    lambda: {
+      memory: 1024,
+      timeout: receiveVerificationLambdaTimeout,
+    },
+    queue: {
+      alarmMaxAgeOfOldestMessage: Duration.hours(1),
+      maxMessageCountAlarmThreshold: 15_000,
+      maxReceiveCount: 3,
+      visibilityTimeout: Duration.seconds(receiveVerificationLambdaTimeout.toSeconds() * 2 + 1),
+      createRetryLambda: false,
+    },
+    eventSource: {
+      batchSize: 1,
+      reportBatchItemFailures: true,
+      maxConcurrency: 4,
+    },
+    waitTime: Duration.seconds(0), // No limit
+  },
+  receiveFlatFileResponse: {
+    name: "SurescriptsReceiveFlatFileResponse",
+    entry: "surescripts-receive-flat-file-response",
+    lambda: {
+      memory: 2048,
+      timeout: receiveFlatFileResponseLambdaTimeout,
+    },
+    queue: {
+      alarmMaxAgeOfOldestMessage: Duration.hours(1),
+      maxMessageCountAlarmThreshold: 15_000,
+      maxReceiveCount: 3,
+      visibilityTimeout: Duration.seconds(receiveFlatFileResponseLambdaTimeout.toSeconds() * 2 + 1),
+      createRetryLambda: false,
+    },
+    eventSource: {
+      batchSize: 1,
+      reportBatchItemFailures: true,
+      maxConcurrency: 4,
+    },
+    waitTime: Duration.seconds(0), // No limit
+  },
+};
+
+function surescriptsEnvironmentVariables({
+  surescripts,
+  allowReplicaAccess = true,
+  allowBundleAccess = false,
+}: {
+  surescripts: EnvConfig["surescripts"];
+  allowReplicaAccess?: boolean;
+  allowBundleAccess?: boolean;
+}): Record<string, string> {
+  return {
+    SURESCRIPTS_SFTP_HOST: surescripts?.surescriptsHost,
+    SURESCRIPTS_SFTP_SENDER_ID: surescripts?.surescriptsSenderId,
+    SURESCRIPTS_SFTP_SENDER_PASSWORD: surescripts?.secrets.SURESCRIPTS_SFTP_SENDER_PASSWORD,
+    SURESCRIPTS_SFTP_RECEIVER_ID: surescripts?.surescriptsReceiverId,
+    SURESCRIPTS_SFTP_PUBLIC_KEY: surescripts?.secrets.SURESCRIPTS_SFTP_PUBLIC_KEY,
+    SURESCRIPTS_SFTP_PRIVATE_KEY: surescripts?.secrets.SURESCRIPTS_SFTP_PRIVATE_KEY,
+    ...(allowReplicaAccess
+      ? { SURESCRIPTS_REPLICA_BUCKET_NAME: surescripts?.surescriptsReplicaBucketName }
+      : {}),
+    ...(allowBundleAccess
+      ? { SURESCRIPTS_BUNDLE_BUCKET_NAME: surescripts?.surescriptsBundleBucketName }
+      : {}),
+  };
+}
+
+interface SurescriptsNestedStackProps extends NestedStackProps {
+  config: EnvConfig;
+  vpc: ec2.IVpc;
+  alarmAction?: SnsAction;
+  lambdaLayers: LambdaLayers;
+  surescriptsReplicaBucket: s3.Bucket;
+  surescriptsBundleBucket: s3.Bucket;
```

Comment:
Why aren't these created on this nested stack? Actually, it gets created in the constructor and we also require on the props? Left-over?

Reply:
Indeed this was left-over and I didn't spot it, thanks for pointing it out! Fixing this now.